File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Double> averageOfLevels(TreeNode root) {
        LinkedList<Double> ans = new LinkedList<>();
        LinkedList<TreeNode> que = new LinkedList<>();
        int level = 0;
        que.addFirst(root);
        
        while(que.size()!=0) {
            int size = que.size();
            double orSize = size;
            double levelAns = 0;
            while(size-- >0) {
                TreeNode rn = que.removeFirst();
                levelAns += rn.val;
                
                if(rn.left != null) {
                    que.addLast(rn.left);
                }
                
                if(rn.right != null) {
                    que.addLast(rn.right);
                }
            }
            ans.addLast(levelAns/orSize);
            level++;
        }
        return ans;
    }
}

File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        vector<double> ans;
        queue<TreeNode*> que;
        que.push(root);
        int level = 0;
        
        while(que.size()!=0) {
            double size = que.size();
            double orSize = size;
            double levelAns = 0;
            while(size-->0) {
                TreeNode* rn = que.front(); que.pop();
                levelAns += rn->val;
                
                if(rn->left != nullptr) {
                    que.push(rn->left);
                }
                
                if(rn->right != nullptr) {
                    que.push(rn->right);
                }
            }
            ans.push_back(levelAns/orSize);
            level++;
        }
        return ans;
    }
};